### PR TITLE
RM-7704 (v2.28): Race Condition in AccessibilityAnalyzer regarding IFrames in Firefox Version 85

### PR DESCRIPTION
**Pull request by raffaelfoidl**
https://re-motion.atlassian.net/browse/RM-7704
Backport of #47

### CODE REVIEW

File: Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs [hunk 1]
```diff
@@ -140,59 +142,68 @@ public void IgnoreRule (AccessibilityRuleID rule)
     /// Analyzes specific <see cref="ControlObject"/> on Webpage.
     /// </summary>
     /// <param name="controlObject"><see cref="ControlObject"/> to analyze.</param>
+    /// <param name="timeout">
+    /// Maximum wait time after which an iframe on the page must be visible to be analyzed. If <see langword="null"/>, a default value of 3000 ms is used.
+    /// </param>
     /// <returns> Result of the analysis.</returns>
     [NotNull]
-    public AccessibilityResult Analyze ([NotNull] ControlObject controlObject)
+    public AccessibilityResult Analyze ([NotNull] ControlObject controlObject, [CanBeNull] TimeSpan? timeout = null)
```
Comment (iam-mholle) [inline Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs:150]: Redundant on a nullable value type.

```suggestion
    public AccessibilityResult Analyze ([NotNull] ControlObject controlObject, TimeSpan? timeout = null)
```

File: Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs [hunk 2]
```diff
@@ -140,59 +142,68 @@ public void IgnoreRule (AccessibilityRuleID rule)
...
     /// Analyzes a specific element on webpage.
     /// </summary>
     /// <param name="cssSelector">CSS selector of the <see cref="ControlObject"/> to analyze.</param>
+    /// <param name="timeout">
+    /// Maximum wait time after which an iframe on the page must be visible to be analyzed. If <see langword="null"/>, a default value of 3000 ms is used.
+    /// </param>
     /// <returns> Result of the analysis.</returns>
     [NotNull]
-    public AccessibilityResult Analyze ([NotNull] string cssSelector)
+    public AccessibilityResult Analyze ([NotNull] string cssSelector, [CanBeNull] TimeSpan? timeout = null)
```
Comment (iam-mholle) [inline Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs:166]: Redundant on a nullable value type.

```suggestion
    public AccessibilityResult Analyze ([NotNull] string cssSelector, TimeSpan? timeout = null)
```

File: Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs [hunk 3]
```diff
@@ -140,59 +142,68 @@ public void IgnoreRule (AccessibilityRuleID rule)
...
+    /// Maximum wait time after which an iframe on the page must be visible to be analyzed. If <see langword="null"/>, a default value of 3000 ms is used.
+    /// </param>
     /// <remarks>
     /// Switches to the outermost frame before the analysis and back to
     /// the original frame after the analysis completed.
     /// </remarks>
     /// <returns> Results of the analysis.</returns>
     [NotNull]
-    public AccessibilityResult Analyze ()
+    public AccessibilityResult Analyze ([CanBeNull] TimeSpan? timeout = null)
```
Comment (iam-mholle) [inline Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs:185]:
```suggestion
    public AccessibilityResult Analyze (TimeSpan? timeout = null)
```

**Review by iam-mholle — APPROVED** [review]

**Comment by iam-mholle** [comment]
Whoops, missed those redundant attributes in #47, they do no harm though.

File: Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs (hunk 2 above)
Comment (MichaelKetting) [reply Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs:166]: good eye. missed that one on develop :o

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**